### ISSUE TITLE
Deleted and re-created namespace cannot be used until tf-controller restart

### ISSUE THREAD

**Issue by greenu** [weaveworks/tf-controller#641]
TLS secret is not regenerating after deleting.

**How to reproduce:**
1. Create a namespace in k8s and a terraform resource in it. Runner runs apply, everything goes well.
2. Delete the terraform resource, Runner runs destoy, everything goes well. TLS secret stays in the namespace
3. Delete the namespace along with TLS secret. Then create it again with the same name. And add a terraform resource in it.
4. Runner goes into CrashLoopBackOff with error `secrets "terraform-runner.tls-1686151888" not found`

Log of tf-controller says that secret is there despite it's not.
```
"msg":"trigger namespace tls secret generation","controller":"terraform"
"logger":"cert-rotation","msg":"TLS already generated for ","namespace":"delete-me"
"msg":"show runner pod state: ","controller":"terraform", ... ... "state":"running"
```

Restart of controller fixes that.

**What I expect:**

TLS is generated when it's not there without restart.

**Comment by yitsushi** [comment]
I can't reproduce this specific error, but I can't say it's resolved, at least the "Delete the namespace" part.

### The simplest approach to the problem to just delete the mtls cert

If I delete the mtls cert, it will create a new one, so that's good. If I create a new Terraform resource, delete the cert, it will create a new one as well.


### Deleted whole namespace

When we install the helm chart, we create a ServiceAccount in all namespaces where the runner is allowed to run. When we delete the namespace, we delete the ServiceAccount too, and when we re-create the namespace, the ServiceAccount will not be there. As a result the runner has no ServiceAccount in that namespace and can't be created.

With the current architecture, we do not support deleting the namespace and we don't have plans to do it directly on that layer.

What's your opinion @chanwit ?

**Comment by yitsushi** [comment]
Related #584

**Comment by chanwit** [comment]
> With the current architecture, we do not support deleting the namespace and we don't have plans to do it directly on that layer.

That's correct. We're working on revisiting the current architecture to enhance it and support resource deletion properly for many complex scenarios, including this one.
